src/container/streamAnalyticsManager.ts

```diff
@@ -4,40 +4,129 @@
+                }
+            } catch (err) {
+                this.asaUpdateStatus = ASAUpdateStatus.Idle;
+                throw err;
+            }
+        }
+    }
+
+    private async updateJobInfo(templateFile: string, moduleName: string): Promise<any> {
+        return await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: "Updating Stream Analytics Job information",
+        }, async (): Promise<object> => {
+            const ASAInfo = await this.getJobInfoFromDeploymentTemplate(templateFile, moduleName);
+            const subscription = await this.getJobSubscription(ASAInfo);
+            const ASAJobResourceId: string = ASAInfo.ASAJobResourceId;
+            return await this.queryASAJobInfo(ASAJobResourceId, subscription.session);
+        });
+    }
+
+    private async isASAJobUpdateAvailable(templateFile: string, moduleName: string): Promise<boolean> {
+        return await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: "Checking configurations update for Stream Analytics Job",
+        }, async (): Promise<boolean> => {
+            const ASAInfo = await this.getJobInfoFromDeploymentTemplate(templateFile, moduleName);
+            const GetASAJobApiUrl: string = `https://management.azure.com${ASAInfo.ASAJobResourceId}?api-version=2019-06-01`;
+            const curEtag = ASAInfo.ASAJobEtag;
+            const subscription = await this.getJobSubscription(ASAInfo);
+            const { aadAccessToken } = await Utility.acquireAadToken(subscription.session);
+            const jobInfo = await request.get(GetASAJobApiUrl, {
+                auth: {
+                    bearer: aadAccessToken,
+                },
+                resolveWithFullResponse: true,
+            });
 
-            const { aadAccessToken } = await Utility.acquireAadToken(streamingJob.azureSubscription.session);
+            const latestETag = jobInfo.headers.etag;
+            return latestETag !== curEtag;
+        });
+    }
+
+    private async updateASAJobInfoModuleTwin(templateFile: string, moduleName: string) {
+        const jobInfo = await this.updateJobInfo(templateFile, moduleName);
+        const moduleTwin = jobInfo.twin.content;
+        const templateJson = await fse.readJson(templateFile);
+        templateJson.modulesContent[moduleName] = moduleTwin;
+        await fse.writeFile(templateFile, JSON.stringify(templateJson, null, 2), { encoding: "utf8" });
+    }
+
+    private async queryASAJobInfo(resourceId: string, session: AzureSession) {
+        try {
+            const apiUrl: string = `https://management.azure.com${resourceId}/publishedgepackage?api-version=2019-06-01`;
+            const { aadAccessToken } = await Utility.acquireAadToken(session);
 
-            const publishResponse = await request.post(publishJobUrl, {
+            const publishResponse = await request.post(apiUrl, {
```

Comment:
Why query use post?

Reply:
The publish api should use post: https://docs.microsoft.com/en-us/rest/api/streamanalytics/stream-analytics-publish-edge-package.
I think I can update this function name to publishAndQueryASAJobInfo, to make it more clear